// @d-id/node-sdk/dist/packages/talks/lib/api/streams-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Video Endpoints
 * Talks
 *
 * The version of the OpenAPI document: 1.1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StreamsApi = exports.StreamsApiFactory = exports.StreamsApiFp = exports.StreamsApiAxiosParamCreator = void 0;
const tslib_1 = require("tslib");
const axios_1 = require("axios");
// URLSearchParams not necessarily used
// @ts-ignore
const url_1 = require("url");
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * StreamsApi - axios parameter creator
 * @export
 */
const StreamsApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Updates the server with a WebRTC ICE candidate. If the candidate parameter is missing or a value of null is given the added ICE candidate is an \"end-of-candidates\" indicator.
         * @summary Submit network information
         * @param {string} id the stream identifier
         * @param {CreateStreamIceCandidateRequest} createStreamIceCandidateRequest
         * @param {string} [cookie]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        addIceCandidate: (id, createStreamIceCandidateRequest, cookie, options = {}) => tslib_1.__awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('addIceCandidate', 'id', id);
            // verify required parameter 'createStreamIceCandidateRequest' is not null or undefined
            (0, common_1.assertParamExists)('addIceCandidate', 'createStreamIceCandidateRequest', createStreamIceCandidateRequest);
            const localVarPath = `/talks/streams/{id}/ice`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new url_1.URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication basic required
            // http basic authentication required
            (0, common_1.setBasicAuthToObject)(localVarRequestOptions, configuration);
            if (cookie != null) {
                localVarHeaderParameter['cookie'] = String(cookie);
            }
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(createStreamIceCandidateRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Initiates the creation of a new WebRTC connection to a browser peer. The endpoint returns the necessary information to set up and manage the connection.
         * @summary Create a new stream
         * @param {InitStreamDto} initStreamDto
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createStream: (initStreamDto, options = {}) => tslib_1.__awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'initStreamDto' is not null or undefined
            (0, common_1.assertParamExists)('createStream', 'initStreamDto', initStreamDto);
            const localVarPath = `/talks/streams`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new url_1.URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication basic required
            // http basic authentication required
            (0, common_1.setBasicAuthToObject)(localVarRequestOptions, configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(initStreamDto, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @summary Create a talk stream
         * @param {string} id the stream identifier
         * @param {CreateTalkStreamRequest} createTalkStreamRequest
         * @param {string} [cookie]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createTalkStream: (id, createTalkStreamRequest, cookie, options = {}) => tslib_1.__awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('createTalkStream', 'id', id);
            // verify required parameter 'createTalkStreamRequest' is not null or undefined
            (0, common_1.assertParamExists)('createTalkStream', 'createTalkStreamRequest', createTalkStreamRequest);
            const localVarPath = `/talks/streams/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new url_1.URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication basic required
            // http basic authentication required
            (0, common_1.setBasicAuthToObject)(localVarRequestOptions, configuration);
            if (cookie != null) {
                localVarHeaderParameter['cookie'] = String(cookie);
            }
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(createTalkStreamRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @summary Delete a talk stream
         * @param {string} id the stream identifier
         * @param {DeleteStreamRequest} deleteStreamRequest
         * @param {string} [cookie]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteStream: (id, deleteStreamRequest, cookie, options = {}) => tslib_1.__awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('deleteStream', 'id', id);
            // verify required parameter 'deleteStreamRequest' is not null or undefined
            (0, common_1.assertParamExists)('deleteStream', 'deleteStreamRequest', deleteStreamRequest);
            const localVarPath = `/talks/streams/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new url_1.URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'DELETE' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication basic required
            // http basic authentication required
            (0, common_1.setBasicAuthToObject)(localVarRequestOptions, configuration);
            if (cookie != null) {
                localVarHeaderParameter['cookie'] = String(cookie);
            }
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(deleteStreamRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         * Updates the server with the SDP answer to the SDP offer received from the initial call to the `/streams` endpoint, during the offer/answer negotiation of a WebRTC connection.
         * @summary Start a stream
         * @param {string} id the stream identifier
         * @param {CreateStreamRequestDto} createStreamRequestDto
         * @param {string} [cookie]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        startConnection: (id, createStreamRequestDto, cookie, options = {}) => tslib_1.__awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('startConnection', 'id', id);
            // verify required parameter 'createStreamRequestDto' is not null or undefined
            (0, common_1.assertParamExists)('startConnection', 'createStreamRequestDto', createStreamRequestDto);
            const localVarPath = `/talks/streams/{id}/sdp`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new url_1.URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication basic required
            // http basic authentication required
            (0, common_1.setBasicAuthToObject)(localVarRequestOptions, configuration);
            if (cookie != null) {
                localVarHeaderParameter['cookie'] = String(cookie);
            }
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(createStreamRequestDto, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
    };
};
exports.StreamsApiAxiosParamCreator = StreamsApiAxiosParamCreator;
/**
 * StreamsApi - functional programming interface
 * @export
 */
const StreamsApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.StreamsApiAxiosParamCreator)(configuration);
    return {
        /**
         * Updates the server with a WebRTC ICE candidate. If the candidate parameter is missing or a value of null is given the added ICE candidate is an \"end-of-candidates\" indicator.
         * @summary Submit network information
         * @param {string} id the stream identifier
         * @param {CreateStreamIceCandidateRequest} createStreamIceCandidateRequest
         * @param {string} [cookie]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        addIceCandidate(id, createStreamIceCandidateRequest, cookie, options) {
            var _a, _b, _c;
            return tslib_1.__awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.addIceCandidate(id, createStreamIceCandidateRequest, cookie, options);
                const index = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const operationBasePath = (_c = (_b = base_1.operationServerMap['StreamsApi.addIceCandidate']) === null || _b === void 0 ? void 0 : _b[index]) === null || _c === void 0 ? void 0 : _c.url;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, operationBasePath || basePath);
            });
        },
        /**
         * Initiates the creation of a new WebRTC connection to a browser peer. The endpoint returns the necessary information to set up and manage the connection.
         * @summary Create a new stream
         * @param {InitStreamDto} initStreamDto
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createStream(initStreamDto, options) {
            var _a, _b, _c;
            return tslib_1.__awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.createStream(initStreamDto, options);
                const index = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const operationBasePath = (_c = (_b = base_1.operationServerMap['StreamsApi.createStream']) === null || _b === void 0 ? void 0 : _b[index]) === null || _c === void 0 ? void 0 : _c.url;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, operationBasePath || basePath);
            });
        },
        /**
         *
         * @summary Create a talk stream
         * @param {string} id the stream identifier
         * @param {CreateTalkStreamRequest} createTalkStreamRequest
         * @param {string} [cookie]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createTalkStream(id, createTalkStreamRequest, cookie, options) {
            var _a, _b, _c;
            return tslib_1.__awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.createTalkStream(id, createTalkStreamRequest, cookie, options);
                const index = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const operationBasePath = (_c = (_b = base_1.operationServerMap['StreamsApi.createTalkStream']) === null || _b === void 0 ? void 0 : _b[index]) === null || _c === void 0 ? void 0 : _c.url;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, operationBasePath || basePath);
            });
        },
        /**
         *
         * @summary Delete a talk stream
         * @param {string} id the stream identifier
         * @param {DeleteStreamRequest} deleteStreamRequest
         * @param {string} [cookie]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteStream(id, deleteStreamRequest, cookie, options) {
            var _a, _b, _c;
            return tslib_1.__awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.deleteStream(id, deleteStreamRequest, cookie, options);
                const index = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const operationBasePath = (_c = (_b = base_1.operationServerMap['StreamsApi.deleteStream']) === null || _b === void 0 ? void 0 : _b[index]) === null || _c === void 0 ? void 0 : _c.url;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, operationBasePath || basePath);
            });
        },
        /**
         * Updates the server with the SDP answer to the SDP offer received from the initial call to the `/streams` endpoint, during the offer/answer negotiation of a WebRTC connection.
         * @summary Start a stream
         * @param {string} id the stream identifier
         * @param {CreateStreamRequestDto} createStreamRequestDto
         * @param {string} [cookie]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        startConnection(id, createStreamRequestDto, cookie, options) {
            var _a, _b, _c;
            return tslib_1.__awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.startConnection(id, createStreamRequestDto, cookie, options);
                const index = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const operationBasePath = (_c = (_b = base_1.operationServerMap['StreamsApi.startConnection']) === null || _b === void 0 ? void 0 : _b[index]) === null || _c === void 0 ? void 0 : _c.url;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, operationBasePath || basePath);
            });
        },
    };
};
exports.StreamsApiFp = StreamsApiFp;
/**
 * StreamsApi - factory interface
 * @export
 */
const StreamsApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.StreamsApiFp)(configuration);
    return {
        /**
         * Updates the server with a WebRTC ICE candidate. If the candidate parameter is missing or a value of null is given the added ICE candidate is an \"end-of-candidates\" indicator.
         * @summary Submit network information
         * @param {string} id the stream identifier
         * @param {CreateStreamIceCandidateRequest} createStreamIceCandidateRequest
         * @param {string} [cookie]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        addIceCandidate(id, createStreamIceCandidateRequest, cookie, options) {
            return localVarFp.addIceCandidate(id, createStreamIceCandidateRequest, cookie, options).then((request) => request(axios, basePath));
        },
        /**
         * Initiates the creation of a new WebRTC connection to a browser peer. The endpoint returns the necessary information to set up and manage the connection.
         * @summary Create a new stream
         * @param {InitStreamDto} initStreamDto
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createStream(initStreamDto, options) {
            return localVarFp.createStream(initStreamDto, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Create a talk stream
         * @param {string} id the stream identifier
         * @param {CreateTalkStreamRequest} createTalkStreamRequest
         * @param {string} [cookie]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createTalkStream(id, createTalkStreamRequest, cookie, options) {
            return localVarFp.createTalkStream(id, createTalkStreamRequest, cookie, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Delete a talk stream
         * @param {string} id the stream identifier
         * @param {DeleteStreamRequest} deleteStreamRequest
         * @param {string} [cookie]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteStream(id, deleteStreamRequest, cookie, options) {
            return localVarFp.deleteStream(id, deleteStreamRequest, cookie, options).then((request) => request(axios, basePath));
        },
        /**
         * Updates the server with the SDP answer to the SDP offer received from the initial call to the `/streams` endpoint, during the offer/answer negotiation of a WebRTC connection.
         * @summary Start a stream
         * @param {string} id the stream identifier
         * @param {CreateStreamRequestDto} createStreamRequestDto
         * @param {string} [cookie]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        startConnection(id, createStreamRequestDto, cookie, options) {
            return localVarFp.startConnection(id, createStreamRequestDto, cookie, options).then((request) => request(axios, basePath));
        },
    };
};
exports.StreamsApiFactory = StreamsApiFactory;
/**
 * StreamsApi - object-oriented interface
 * @export
 * @class StreamsApi
 * @extends {BaseAPI}
 */
class StreamsApi extends base_1.BaseAPI {
    /**
     * Updates the server with a WebRTC ICE candidate. If the candidate parameter is missing or a value of null is given the added ICE candidate is an \"end-of-candidates\" indicator.
     * @summary Submit network information
     * @param {string} id the stream identifier
     * @param {CreateStreamIceCandidateRequest} createStreamIceCandidateRequest
     * @param {string} [cookie]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StreamsApi
     */
    addIceCandidate(id, createStreamIceCandidateRequest, cookie, options) {
        return (0, exports.StreamsApiFp)(this.configuration).addIceCandidate(id, createStreamIceCandidateRequest, cookie, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Initiates the creation of a new WebRTC connection to a browser peer. The endpoint returns the necessary information to set up and manage the connection.
     * @summary Create a new stream
     * @param {InitStreamDto} initStreamDto
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StreamsApi
     */
    createStream(initStreamDto, options) {
        return (0, exports.StreamsApiFp)(this.configuration).createStream(initStreamDto, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Create a talk stream
     * @param {string} id the stream identifier
     * @param {CreateTalkStreamRequest} createTalkStreamRequest
     * @param {string} [cookie]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StreamsApi
     */
    createTalkStream(id, createTalkStreamRequest, cookie, options) {
        return (0, exports.StreamsApiFp)(this.configuration).createTalkStream(id, createTalkStreamRequest, cookie, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Delete a talk stream
     * @param {string} id the stream identifier
     * @param {DeleteStreamRequest} deleteStreamRequest
     * @param {string} [cookie]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StreamsApi
     */
    deleteStream(id, deleteStreamRequest, cookie, options) {
        return (0, exports.StreamsApiFp)(this.configuration).deleteStream(id, deleteStreamRequest, cookie, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Updates the server with the SDP answer to the SDP offer received from the initial call to the `/streams` endpoint, during the offer/answer negotiation of a WebRTC connection.
     * @summary Start a stream
     * @param {string} id the stream identifier
     * @param {CreateStreamRequestDto} createStreamRequestDto
     * @param {string} [cookie]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof StreamsApi
     */
    startConnection(id, createStreamRequestDto, cookie, options) {
        return (0, exports.StreamsApiFp)(this.configuration).startConnection(id, createStreamRequestDto, cookie, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.StreamsApi = StreamsApi;
