// @d-id/node-sdk/dist/packages/talks/lib/api/talks-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Video Endpoints
 * Talks
 *
 * The version of the OpenAPI document: 1.1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TalksApi = exports.TalksApiFactory = exports.TalksApiFp = exports.TalksApiAxiosParamCreator = void 0;
const tslib_1 = require("tslib");
const axios_1 = require("axios");
// URLSearchParams not necessarily used
// @ts-ignore
const url_1 = require("url");
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * TalksApi - axios parameter creator
 * @export
 */
const TalksApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @summary Create a talk
         * @param {CreateTalkRequest} createTalkRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createTalk: (createTalkRequest, options = {}) => tslib_1.__awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'createTalkRequest' is not null or undefined
            (0, common_1.assertParamExists)('createTalk', 'createTalkRequest', createTalkRequest);
            const localVarPath = `/talks`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new url_1.URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication basic required
            // http basic authentication required
            (0, common_1.setBasicAuthToObject)(localVarRequestOptions, configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(createTalkRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @summary Delete a specific talk
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteTalk: (id, options = {}) => tslib_1.__awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('deleteTalk', 'id', id);
            const localVarPath = `/talks/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new url_1.URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'DELETE' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication basic required
            // http basic authentication required
            (0, common_1.setBasicAuthToObject)(localVarRequestOptions, configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @summary Get a specific talk
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getTalk: (id, options = {}) => tslib_1.__awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('getTalk', 'id', id);
            const localVarPath = `/talks/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new url_1.URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication basic required
            // http basic authentication required
            (0, common_1.setBasicAuthToObject)(localVarRequestOptions, configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @summary Get talks
         * @param {number} [limit] pagination - the amount of talks to return
         * @param {string} [token] pagination - the pagination token from the previous response, default &#x3D; 100
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getTalks: (limit, token, options = {}) => tslib_1.__awaiter(this, void 0, void 0, function* () {
            const localVarPath = `/talks`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new url_1.URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication basic required
            // http basic authentication required
            (0, common_1.setBasicAuthToObject)(localVarRequestOptions, configuration);
            if (limit !== undefined) {
                localVarQueryParameter['limit'] = limit;
            }
            if (token !== undefined) {
                localVarQueryParameter['token'] = token;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
    };
};
exports.TalksApiAxiosParamCreator = TalksApiAxiosParamCreator;
/**
 * TalksApi - functional programming interface
 * @export
 */
const TalksApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.TalksApiAxiosParamCreator)(configuration);
    return {
        /**
         *
         * @summary Create a talk
         * @param {CreateTalkRequest} createTalkRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createTalk(createTalkRequest, options) {
            var _a, _b, _c;
            return tslib_1.__awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.createTalk(createTalkRequest, options);
                const index = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const operationBasePath = (_c = (_b = base_1.operationServerMap['TalksApi.createTalk']) === null || _b === void 0 ? void 0 : _b[index]) === null || _c === void 0 ? void 0 : _c.url;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, operationBasePath || basePath);
            });
        },
        /**
         *
         * @summary Delete a specific talk
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteTalk(id, options) {
            var _a, _b, _c;
            return tslib_1.__awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.deleteTalk(id, options);
                const index = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const operationBasePath = (_c = (_b = base_1.operationServerMap['TalksApi.deleteTalk']) === null || _b === void 0 ? void 0 : _b[index]) === null || _c === void 0 ? void 0 : _c.url;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, operationBasePath || basePath);
            });
        },
        /**
         *
         * @summary Get a specific talk
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getTalk(id, options) {
            var _a, _b, _c;
            return tslib_1.__awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getTalk(id, options);
                const index = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const operationBasePath = (_c = (_b = base_1.operationServerMap['TalksApi.getTalk']) === null || _b === void 0 ? void 0 : _b[index]) === null || _c === void 0 ? void 0 : _c.url;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, operationBasePath || basePath);
            });
        },
        /**
         *
         * @summary Get talks
         * @param {number} [limit] pagination - the amount of talks to return
         * @param {string} [token] pagination - the pagination token from the previous response, default &#x3D; 100
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getTalks(limit, token, options) {
            var _a, _b, _c;
            return tslib_1.__awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getTalks(limit, token, options);
                const index = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const operationBasePath = (_c = (_b = base_1.operationServerMap['TalksApi.getTalks']) === null || _b === void 0 ? void 0 : _b[index]) === null || _c === void 0 ? void 0 : _c.url;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, operationBasePath || basePath);
            });
        },
    };
};
exports.TalksApiFp = TalksApiFp;
/**
 * TalksApi - factory interface
 * @export
 */
const TalksApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.TalksApiFp)(configuration);
    return {
        /**
         *
         * @summary Create a talk
         * @param {CreateTalkRequest} createTalkRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createTalk(createTalkRequest, options) {
            return localVarFp.createTalk(createTalkRequest, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Delete a specific talk
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteTalk(id, options) {
            return localVarFp.deleteTalk(id, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get a specific talk
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getTalk(id, options) {
            return localVarFp.getTalk(id, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get talks
         * @param {number} [limit] pagination - the amount of talks to return
         * @param {string} [token] pagination - the pagination token from the previous response, default &#x3D; 100
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getTalks(limit, token, options) {
            return localVarFp.getTalks(limit, token, options).then((request) => request(axios, basePath));
        },
    };
};
exports.TalksApiFactory = TalksApiFactory;
/**
 * TalksApi - object-oriented interface
 * @export
 * @class TalksApi
 * @extends {BaseAPI}
 */
class TalksApi extends base_1.BaseAPI {
    /**
     *
     * @summary Create a talk
     * @param {CreateTalkRequest} createTalkRequest
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof TalksApi
     */
    createTalk(createTalkRequest, options) {
        return (0, exports.TalksApiFp)(this.configuration).createTalk(createTalkRequest, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Delete a specific talk
     * @param {string} id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof TalksApi
     */
    deleteTalk(id, options) {
        return (0, exports.TalksApiFp)(this.configuration).deleteTalk(id, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get a specific talk
     * @param {string} id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof TalksApi
     */
    getTalk(id, options) {
        return (0, exports.TalksApiFp)(this.configuration).getTalk(id, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get talks
     * @param {number} [limit] pagination - the amount of talks to return
     * @param {string} [token] pagination - the pagination token from the previous response, default &#x3D; 100
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof TalksApi
     */
    getTalks(limit, token, options) {
        return (0, exports.TalksApiFp)(this.configuration).getTalks(limit, token, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.TalksApi = TalksApi;
