// @d-id/node-sdk/dist/packages/clips/client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ClipsClient = void 0;
const tslib_1 = require("tslib");
const lib_1 = require("./lib");
const client_base_1 = require("../../client-base");
class ClipsClient extends client_base_1.default {
    /**
     *
     * @param configuration
     */
    constructor(configuration, ws = null) {
        super(ws);
        this.clipsApi = new lib_1.ClipsApi(configuration);
    }
    handleWsMessage(message) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            try {
                const data = JSON.parse(message.toString());
                const subscription = this.subscriptions.get(data.id);
                if (subscription) {
                    if (data.type === "clip/done") {
                        const clip = yield this.get(data.id);
                        subscription.resolve(clip);
                    }
                    if (data.type === "clip/error") {
                        subscription.reject(data.data);
                    }
                    this.subscriptions.delete(data.id);
                }
            }
            catch (err) {
                console.error("Error handling WS message:", err);
            }
        });
    }
    /**
     * Get clips
     * @param limit
     * @param token
     * @returns
     */
    list(limit, token) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const clipsResponse = yield this.clipsApi.getClips(limit, token);
            return clipsResponse.data;
        });
    }
    /**
     * Get a specific clip
     * @param id
     * @returns
     */
    get(id) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const clipResponse = yield this.clipsApi.getClip(id);
            return clipResponse.data;
        });
    }
    /**
     * Create a clip
     * @param clipRequest
     * @returns
     */
    create(clipRequest) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const createClipResponse = yield this.clipsApi.createClip(clipRequest);
            return createClipResponse.data;
        });
    }
    /**
     * Create a clip async.
     * This method will return a promise that will resolve when the clip is ready.
     * @param clipRequest
     * @returns
     */
    createAsync(clipRequest) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            if (!this.ws) {
                throw new Error("WebSocket connection is not open");
            }
            const createClipResponse = yield this.clipsApi.createClip(clipRequest);
            return new Promise((resolve, reject) => {
                this.subscriptions.set(createClipResponse.data.id, { resolve, reject });
            });
        });
    }
    /**
     * Delete a specific clip
     * @param id
     */
    delete(id) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const deleteClipResponse = yield this.clipsApi.deleteClip(id);
            return deleteClipResponse.data;
        });
    }
    /**
     * Get a list of the presenters
     * @param limit
     * @param token
     * @returns
     */
    listPresenters(limit, token) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const clipPresentersResponse = yield this.clipsApi.getPresenters(limit, token);
            return clipPresentersResponse.data;
        });
    }
    /**
     * Get presenter by id
     * @param id
     * @returns
     */
    getPresenter(id) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const clipPresenterResponse = yield this.clipsApi.getPresenterById(id);
            return clipPresenterResponse.data;
        });
    }
    /**
     * Get presenter drivers by id
     * @param id
     * @returns
     */
    getPresenterDrivers(id) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const clipPresenterDriversResponse = yield this.clipsApi.getPresenterDriversById(id);
            return clipPresenterDriversResponse.data;
        });
    }
}
exports.ClipsClient = ClipsClient;
