// @d-id/node-sdk/dist/packages/talks/client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TalksClient = void 0;
const tslib_1 = require("tslib");
const lib_1 = require("./lib");
const client_base_1 = require("../../client-base");
/**
 * Talks client
 */
class TalksClient extends client_base_1.default {
    /**
     *
     * @param configuration
     */
    constructor(configuration, ws = null) {
        super(ws);
        this.talksApi = new lib_1.TalksApi(configuration);
    }
    handleWsMessage(message) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            console.log("Handling WS message", message);
            try {
                const data = JSON.parse(message.toString());
                console.log("Handling WS message", data);
                const subscription = this.subscriptions.get(data.id);
                if (subscription) {
                    if (data.type === "talk/done") {
                        const talk = yield this.get(data.id);
                        subscription.resolve(talk);
                    }
                    if (data.type === "talk/error") {
                        subscription.reject(data.data);
                    }
                    this.subscriptions.delete(data.id);
                }
            }
            catch (err) {
                console.error("Error handling WS message:", err);
            }
        });
    }
    /**
     *
     * @summary Get talks
     * @param {number} [limit] pagination - the amount of talks to return
     * @param {string} [token] pagination - the pagination token from the previous response, default &#x3D; 100
     */
    list(limit, token) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const talksResponse = yield this.talksApi.getTalks(limit, token);
            return talksResponse.data;
        });
    }
    /**
     *
     * @param id
     * @returns
     */
    get(id) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const talkResponse = yield this.talksApi.getTalk(id);
            return talkResponse.data;
        });
    }
    /**
     *
     * @param talkRequest
     * @returns
     */
    create(talkRequest) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const createTalkResponse = yield this.talksApi.createTalk(talkRequest);
            return createTalkResponse.data;
        });
    }
    /**
     *
     * @param talkRequest
     * @returns
     */
    createAsync(talkRequest) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            if (!this.ws) {
                throw new Error("WebSocket connection is not open");
            }
            console.log("Creating talk async");
            const createTalkResponse = yield this.talksApi.createTalk(talkRequest);
            console.log("CREATED");
            return new Promise((resolve, reject) => {
                console.log("IN this PROMISE", createTalkResponse.data.id);
                this.subscriptions.set(createTalkResponse.data.id, { resolve, reject });
            });
        });
    }
    /**
     *
     * @param id
     * @returns
     */
    delete(id) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const deleteTalkResponse = yield this.talksApi.deleteTalk(id);
            return deleteTalkResponse.data;
        });
    }
    /**
     * Init a stream
     *
     * @summary Initiates the creation of a new WebRTC connection to a browser peer.
     * The endpoint returns the necessary information to set up and manage the connection.
     * @param initStreamParams
     * @returns
     */
    initStream(initStreamParams) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const streamResponse = yield this.streamsApi.createStream(initStreamParams);
            return streamResponse.data;
        });
    }
    /**
     * Start a stream
     *
     * @summary Updates the server with the SDP answer to the SDP offer received from the initial call to the /streams endpoint,
     * during the offer/answer negotiation of a WebRTC connection.
     * @param id
     * @param starStreamRequest
     * @returns
     */
    startStream(id, starStreamRequest) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const streamResponse = yield this.streamsApi.startConnection(id, starStreamRequest);
            return streamResponse.data;
        });
    }
    /**
     * Submit network
     *
     * @summary Updates the server with a WebRTC ICE candidate.
     * If the candidate parameter is missing or a value of null is given
     * the added ICE candidate is an "end-of-candidates" indicator.
     * @param id
     * @param submitNetworkRequest
     * @returns
     */
    submitNetwork(id, submitNetworkRequest) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const streamResponse = yield this.streamsApi.addIceCandidate(id, submitNetworkRequest);
            return streamResponse.data;
        });
    }
    /**
     * Create a talk stream
     *
     * @param id
     * @param createTalkStreamRequest
     * @returns
     */
    createStream(id, createTalkStreamRequest) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const streamResponse = yield this.streamsApi.createTalkStream(id, createTalkStreamRequest);
            return streamResponse.data;
        });
    }
    /**
     * Delete a talk stream
     *
     * @param id
     * @param deleteRequest
     * @returns
     */
    deleteStream(id, deleteRequest) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            const streamResponse = yield this.streamsApi.deleteStream(id, deleteRequest);
            return streamResponse.data;
        });
    }
}
exports.TalksClient = TalksClient;
