// @d-id/node-sdk/dist/services/socket-service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SocketService = void 0;
/**
 * SocketService
 * Incapsulates the websocket business logic
 */
const ws_1 = require("ws");
/**
 * Default socket url
 * prod - wss://notifications.d-id.com
 * dev - wss://notifications-dev.d-id.com
 * @type {string}
 */
// const DEFAULT_SOKET_URL = "wss://notifications.d-id.com";
const DEFAULT_SOKET_URL = "wss://notifications-dev.d-id.com";
/**
 * SocketService
 */
class SocketService {
    constructor(apiKey) {
        /**
         * @property socket
         */
        this.socket = null;
        this.apiKey = apiKey;
        this.ensureWebSocketConnection();
    }
    /**
     *
     * @param apiKey
     * @returns
     */
    static getInstance(apiKey) {
        if (this.instance === null) {
            this.instance = new SocketService(apiKey);
        }
        return this.instance;
    }
    /**
     * Get connection params
     * @private
     * @returns
     */
    getConnectionParams() {
        const authParams = new URLSearchParams();
        authParams.set("authorization", `Basic ${this.apiKey}`);
        return authParams;
    }
    /**
     * Ensure that the websocket connection is open
     */
    ensureWebSocketConnection() {
        if (!this.socket || this.socket.readyState === ws_1.WebSocket.CLOSED) {
            try {
                // console.log("Creating websocket connection");
                const authParams = this.getConnectionParams();
                const socketUrl = `${DEFAULT_SOKET_URL}?${authParams.toString()}`;
                this.socket = new ws_1.WebSocket(socketUrl);
                // this.socket.on("open", () => {
                //   console.log("Websocket connection opened");
                // });
            }
            catch (err) {
                console.error("Error creating websocket connection:", err);
            }
        }
    }
    /**
     *
     * @param event
     * @param callback
     */
    on(event, callback) {
        var _a;
        console.log("Setting up WS event listener", event);
        (_a = this.socket) === null || _a === void 0 ? void 0 : _a.on(event, callback);
        this.socket.onmessage = (event) => {
            console.log("WS message", event.data);
        };
        // debug
        this.socket.on("ping", () => {
            console.log("Ping");
        });
        this.socket.on("close", () => {
            console.log("Close");
        });
        this.socket.on("error", () => {
            console.log("Error");
        });
    }
}
exports.SocketService = SocketService;
/**
 * @property instance
 */
SocketService.instance = null;
