// @d-id/node-sdk/dist/did.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const socket_service_1 = require("./services/socket-service");
const talks_1 = require("./packages/talks");
const clips_1 = require("./packages/clips");
const lib_1 = require("./packages/talks/lib");
/**
 * D-ID SDK client
 */
class DidClient {
    /**
     * @param configuration - D-ID sdk client configuration object
     */
    constructor(apiKey, basePath = "https://api.d-id.com") {
        this.config = new lib_1.Configuration({
            basePath,
            baseOptions: {
                headers: {
                    Authorization: `Basic ${apiKey}`,
                    "X-Origin": "sdk-nodejs",
                },
            },
        });
        const socketService = socket_service_1.SocketService.getInstance(apiKey);
        this.talksClient = new talks_1.TalksClient(this.config, socketService);
        this.clipsClient = new clips_1.ClipsClient(this.config, socketService);
    }
    /**
     * Get access to the talks client api wrapper
     */
    get talks() {
        return this.talksClient;
    }
    get clips() {
        return this.clipsClient;
    }
}
exports.default = DidClient;
