// @d-id/node-sdk/dist/socket-service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SocketService = void 0;
const ws_1 = require("ws");
const DEFAULT_SOKET_URL = "wss://notifications-dev.d-id.com";
class SocketService {
    constructor(apiKey) {
        this.socket = null;
        this.apiKey = apiKey;
        this.ensureWebSocketConnection();
    }
    static getInstance(apiKey) {
        if (this.instance === null) {
            this.instance = new SocketService(apiKey);
        }
        return this.instance;
    }
    getConnectionParams() {
        const authParams = new URLSearchParams();
        authParams.set("authorization", `Bearer ${this.apiKey}`);
        return authParams;
    }
    /**
     * Ensure that the websocket connection is open
     */
    ensureWebSocketConnection() {
        if (!this.socket || this.socket.readyState === ws_1.WebSocket.CLOSED) {
            try {
                const authParams = this.getConnectionParams();
                this.socket = new ws_1.WebSocket(`${DEFAULT_SOKET_URL}?${authParams.toString()}`);
            }
            catch (err) {
                console.error("Error creating websocket connection:", err);
            }
        }
    }
    on(event, callback) {
        var _a;
        (_a = this.socket) === null || _a === void 0 ? void 0 : _a.on(event, callback);
    }
}
exports.SocketService = SocketService;
SocketService.instance = null;
