// @d-id/node-sdk/dist/packages/talks/lib/models/voice-ids-amazon.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Video Endpoints
 * Talks
 *
 * The version of the OpenAPI document: 1.1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.VoiceIdsAmazon = void 0;
/**
 *
 * @export
 * @enum {string}
 */
var VoiceIdsAmazon;
(function (VoiceIdsAmazon) {
    VoiceIdsAmazon["Amy"] = "Amy";
    VoiceIdsAmazon["Aria"] = "Aria";
    VoiceIdsAmazon["Ayanda"] = "Ayanda";
    VoiceIdsAmazon["Bianca"] = "Bianca";
    VoiceIdsAmazon["Brian"] = "Brian";
    VoiceIdsAmazon["Camila"] = "Camila";
    VoiceIdsAmazon["Emma"] = "Emma";
    VoiceIdsAmazon["Gabrielle"] = "Gabrielle";
    VoiceIdsAmazon["Ivy"] = "Ivy";
    VoiceIdsAmazon["Joanna"] = "Joanna";
    VoiceIdsAmazon["Joey"] = "Joey";
    VoiceIdsAmazon["Justin"] = "Justin";
    VoiceIdsAmazon["Kendra"] = "Kendra";
    VoiceIdsAmazon["Kevin"] = "Kevin";
    VoiceIdsAmazon["Kimberly"] = "Kimberly";
    VoiceIdsAmazon["Lea"] = "Lea";
    VoiceIdsAmazon["Lucia"] = "Lucia";
    VoiceIdsAmazon["Lupe"] = "Lupe";
    VoiceIdsAmazon["Matthew"] = "Matthew";
    VoiceIdsAmazon["Olivia"] = "Olivia";
    VoiceIdsAmazon["Salli"] = "Salli";
    VoiceIdsAmazon["Seoyeon"] = "Seoyeon";
    VoiceIdsAmazon["Takumi"] = "Takumi";
    VoiceIdsAmazon["Vicki"] = "Vicki";
})(VoiceIdsAmazon || (exports.VoiceIdsAmazon = VoiceIdsAmazon = {}));
