// @d-id/node-sdk/dist/packages/clips/lib/api/clips-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Video Endpoints
 * Clips
 *
 * The version of the OpenAPI document: 1.1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ClipsApi = exports.ClipsApiFactory = exports.ClipsApiFp = exports.ClipsApiAxiosParamCreator = void 0;
const tslib_1 = require("tslib");
const axios_1 = require("axios");
// URLSearchParams not necessarily used
// @ts-ignore
const url_1 = require("url");
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * ClipsApi - axios parameter creator
 * @export
 */
const ClipsApiAxiosParamCreator = function (configuration) {
    return {
        /**
         *
         * @summary Create a clip
         * @param {CreateClipRequest} createClipRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createClip: (createClipRequest, options = {}) => tslib_1.__awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'createClipRequest' is not null or undefined
            (0, common_1.assertParamExists)('createClip', 'createClipRequest', createClipRequest);
            const localVarPath = `/clips`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new url_1.URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication basic required
            // http basic authentication required
            (0, common_1.setBasicAuthToObject)(localVarRequestOptions, configuration);
            localVarHeaderParameter['Content-Type'] = 'application/json';
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = (0, common_1.serializeDataIfNeeded)(createClipRequest, localVarRequestOptions, configuration);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @summary Delete a specific clip
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteClip: (id, options = {}) => tslib_1.__awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('deleteClip', 'id', id);
            const localVarPath = `/clips/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new url_1.URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'DELETE' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication basic required
            // http basic authentication required
            (0, common_1.setBasicAuthToObject)(localVarRequestOptions, configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @summary Get driver list of an actor
         * @param {string} id
         * @param {number} [limit] pagination - the amount of actors to return
         * @param {string} [token] pagination - the pagination token from the previous response, default &#x3D; 100
         * @param {*} [options] Override http request option.
         * @deprecated
         * @throws {RequiredError}
         */
        getActorDrivers: (id, limit, token, options = {}) => tslib_1.__awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('getActorDrivers', 'id', id);
            const localVarPath = `/clips/actors/{id}/drivers`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new url_1.URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (limit !== undefined) {
                localVarQueryParameter['limit'] = limit;
            }
            if (token !== undefined) {
                localVarQueryParameter['token'] = token;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @summary Get actors list
         * @param {number} [limit] pagination - the amount of actors to return
         * @param {string} [token] pagination - the pagination token from the previous response, default &#x3D; 100
         * @param {*} [options] Override http request option.
         * @deprecated
         * @throws {RequiredError}
         */
        getActors: (limit, token, options = {}) => tslib_1.__awaiter(this, void 0, void 0, function* () {
            const localVarPath = `/clips/actors`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new url_1.URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (limit !== undefined) {
                localVarQueryParameter['limit'] = limit;
            }
            if (token !== undefined) {
                localVarQueryParameter['token'] = token;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @summary Get a specific clip
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getClip: (id, options = {}) => tslib_1.__awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('getClip', 'id', id);
            const localVarPath = `/clips/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new url_1.URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication basic required
            // http basic authentication required
            (0, common_1.setBasicAuthToObject)(localVarRequestOptions, configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @summary Get clips
         * @param {number} [limit] pagination - the amount of clips to return
         * @param {string} [token] pagination - the pagination token from the previous response, default &#x3D; 100
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getClips: (limit, token, options = {}) => tslib_1.__awaiter(this, void 0, void 0, function* () {
            const localVarPath = `/clips`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new url_1.URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication basic required
            // http basic authentication required
            (0, common_1.setBasicAuthToObject)(localVarRequestOptions, configuration);
            if (limit !== undefined) {
                localVarQueryParameter['limit'] = limit;
            }
            if (token !== undefined) {
                localVarQueryParameter['token'] = token;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @summary Get presenter by id
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getPresenterById: (id, options = {}) => tslib_1.__awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('getPresenterById', 'id', id);
            const localVarPath = `/clips/presenters/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new url_1.URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication basic required
            // http basic authentication required
            (0, common_1.setBasicAuthToObject)(localVarRequestOptions, configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @summary Get presenter by id
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getPresenterById_1: (id, options = {}) => tslib_1.__awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('getPresenterById_1', 'id', id);
            const localVarPath = `/clips/presenters/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new url_1.URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication basic required
            // http basic authentication required
            (0, common_1.setBasicAuthToObject)(localVarRequestOptions, configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @summary Get presenter drivers by id
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getPresenterDriversById: (id, options = {}) => tslib_1.__awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('getPresenterDriversById', 'id', id);
            const localVarPath = `/clips/presenters/{id}/drivers`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new url_1.URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication basic required
            // http basic authentication required
            (0, common_1.setBasicAuthToObject)(localVarRequestOptions, configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @summary Get presenter drivers by id
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getPresenterDriversById_2: (id, options = {}) => tslib_1.__awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'id' is not null or undefined
            (0, common_1.assertParamExists)('getPresenterDriversById_2', 'id', id);
            const localVarPath = `/clips/presenters/{id}/drivers`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new url_1.URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication basic required
            // http basic authentication required
            (0, common_1.setBasicAuthToObject)(localVarRequestOptions, configuration);
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @summary Get a list of the presenters
         * @param {number} [limit] pagination - the amount of presenters to return
         * @param {string} [token] pagination - the pagination token from the previous response, default &#x3D; 100
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getPresenters: (limit, token, options = {}) => tslib_1.__awaiter(this, void 0, void 0, function* () {
            const localVarPath = `/clips/presenters`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new url_1.URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication basic required
            // http basic authentication required
            (0, common_1.setBasicAuthToObject)(localVarRequestOptions, configuration);
            if (limit !== undefined) {
                localVarQueryParameter['limit'] = limit;
            }
            if (token !== undefined) {
                localVarQueryParameter['token'] = token;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
        /**
         *
         * @summary Get a list of the presenters
         * @param {number} [limit] pagination - the amount of presenters to return
         * @param {string} [token] pagination - the pagination token from the previous response, default &#x3D; 100
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getPresenters_3: (limit, token, options = {}) => tslib_1.__awaiter(this, void 0, void 0, function* () {
            const localVarPath = `/clips/presenters`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new url_1.URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            // authentication bearer required
            // http bearer authentication required
            yield (0, common_1.setBearerAuthToObject)(localVarHeaderParameter, configuration);
            // authentication basic required
            // http basic authentication required
            (0, common_1.setBasicAuthToObject)(localVarRequestOptions, configuration);
            if (limit !== undefined) {
                localVarQueryParameter['limit'] = limit;
            }
            if (token !== undefined) {
                localVarQueryParameter['token'] = token;
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
    };
};
exports.ClipsApiAxiosParamCreator = ClipsApiAxiosParamCreator;
/**
 * ClipsApi - functional programming interface
 * @export
 */
const ClipsApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.ClipsApiAxiosParamCreator)(configuration);
    return {
        /**
         *
         * @summary Create a clip
         * @param {CreateClipRequest} createClipRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createClip(createClipRequest, options) {
            var _a, _b, _c;
            return tslib_1.__awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.createClip(createClipRequest, options);
                const index = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const operationBasePath = (_c = (_b = base_1.operationServerMap['ClipsApi.createClip']) === null || _b === void 0 ? void 0 : _b[index]) === null || _c === void 0 ? void 0 : _c.url;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, operationBasePath || basePath);
            });
        },
        /**
         *
         * @summary Delete a specific clip
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteClip(id, options) {
            var _a, _b, _c;
            return tslib_1.__awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.deleteClip(id, options);
                const index = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const operationBasePath = (_c = (_b = base_1.operationServerMap['ClipsApi.deleteClip']) === null || _b === void 0 ? void 0 : _b[index]) === null || _c === void 0 ? void 0 : _c.url;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, operationBasePath || basePath);
            });
        },
        /**
         *
         * @summary Get driver list of an actor
         * @param {string} id
         * @param {number} [limit] pagination - the amount of actors to return
         * @param {string} [token] pagination - the pagination token from the previous response, default &#x3D; 100
         * @param {*} [options] Override http request option.
         * @deprecated
         * @throws {RequiredError}
         */
        getActorDrivers(id, limit, token, options) {
            var _a, _b, _c;
            return tslib_1.__awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getActorDrivers(id, limit, token, options);
                const index = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const operationBasePath = (_c = (_b = base_1.operationServerMap['ClipsApi.getActorDrivers']) === null || _b === void 0 ? void 0 : _b[index]) === null || _c === void 0 ? void 0 : _c.url;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, operationBasePath || basePath);
            });
        },
        /**
         *
         * @summary Get actors list
         * @param {number} [limit] pagination - the amount of actors to return
         * @param {string} [token] pagination - the pagination token from the previous response, default &#x3D; 100
         * @param {*} [options] Override http request option.
         * @deprecated
         * @throws {RequiredError}
         */
        getActors(limit, token, options) {
            var _a, _b, _c;
            return tslib_1.__awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getActors(limit, token, options);
                const index = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const operationBasePath = (_c = (_b = base_1.operationServerMap['ClipsApi.getActors']) === null || _b === void 0 ? void 0 : _b[index]) === null || _c === void 0 ? void 0 : _c.url;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, operationBasePath || basePath);
            });
        },
        /**
         *
         * @summary Get a specific clip
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getClip(id, options) {
            var _a, _b, _c;
            return tslib_1.__awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getClip(id, options);
                const index = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const operationBasePath = (_c = (_b = base_1.operationServerMap['ClipsApi.getClip']) === null || _b === void 0 ? void 0 : _b[index]) === null || _c === void 0 ? void 0 : _c.url;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, operationBasePath || basePath);
            });
        },
        /**
         *
         * @summary Get clips
         * @param {number} [limit] pagination - the amount of clips to return
         * @param {string} [token] pagination - the pagination token from the previous response, default &#x3D; 100
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getClips(limit, token, options) {
            var _a, _b, _c;
            return tslib_1.__awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getClips(limit, token, options);
                const index = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const operationBasePath = (_c = (_b = base_1.operationServerMap['ClipsApi.getClips']) === null || _b === void 0 ? void 0 : _b[index]) === null || _c === void 0 ? void 0 : _c.url;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, operationBasePath || basePath);
            });
        },
        /**
         *
         * @summary Get presenter by id
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getPresenterById(id, options) {
            var _a, _b, _c;
            return tslib_1.__awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getPresenterById(id, options);
                const index = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const operationBasePath = (_c = (_b = base_1.operationServerMap['ClipsApi.getPresenterById']) === null || _b === void 0 ? void 0 : _b[index]) === null || _c === void 0 ? void 0 : _c.url;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, operationBasePath || basePath);
            });
        },
        /**
         *
         * @summary Get presenter by id
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getPresenterById_1(id, options) {
            var _a, _b, _c;
            return tslib_1.__awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getPresenterById_1(id, options);
                const index = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const operationBasePath = (_c = (_b = base_1.operationServerMap['ClipsApi.getPresenterById_1']) === null || _b === void 0 ? void 0 : _b[index]) === null || _c === void 0 ? void 0 : _c.url;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, operationBasePath || basePath);
            });
        },
        /**
         *
         * @summary Get presenter drivers by id
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getPresenterDriversById(id, options) {
            var _a, _b, _c;
            return tslib_1.__awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getPresenterDriversById(id, options);
                const index = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const operationBasePath = (_c = (_b = base_1.operationServerMap['ClipsApi.getPresenterDriversById']) === null || _b === void 0 ? void 0 : _b[index]) === null || _c === void 0 ? void 0 : _c.url;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, operationBasePath || basePath);
            });
        },
        /**
         *
         * @summary Get presenter drivers by id
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getPresenterDriversById_2(id, options) {
            var _a, _b, _c;
            return tslib_1.__awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getPresenterDriversById_2(id, options);
                const index = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const operationBasePath = (_c = (_b = base_1.operationServerMap['ClipsApi.getPresenterDriversById_2']) === null || _b === void 0 ? void 0 : _b[index]) === null || _c === void 0 ? void 0 : _c.url;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, operationBasePath || basePath);
            });
        },
        /**
         *
         * @summary Get a list of the presenters
         * @param {number} [limit] pagination - the amount of presenters to return
         * @param {string} [token] pagination - the pagination token from the previous response, default &#x3D; 100
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getPresenters(limit, token, options) {
            var _a, _b, _c;
            return tslib_1.__awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getPresenters(limit, token, options);
                const index = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const operationBasePath = (_c = (_b = base_1.operationServerMap['ClipsApi.getPresenters']) === null || _b === void 0 ? void 0 : _b[index]) === null || _c === void 0 ? void 0 : _c.url;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, operationBasePath || basePath);
            });
        },
        /**
         *
         * @summary Get a list of the presenters
         * @param {number} [limit] pagination - the amount of presenters to return
         * @param {string} [token] pagination - the pagination token from the previous response, default &#x3D; 100
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getPresenters_3(limit, token, options) {
            var _a, _b, _c;
            return tslib_1.__awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.getPresenters_3(limit, token, options);
                const index = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const operationBasePath = (_c = (_b = base_1.operationServerMap['ClipsApi.getPresenters_3']) === null || _b === void 0 ? void 0 : _b[index]) === null || _c === void 0 ? void 0 : _c.url;
                return (axios, basePath) => (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration)(axios, operationBasePath || basePath);
            });
        },
    };
};
exports.ClipsApiFp = ClipsApiFp;
/**
 * ClipsApi - factory interface
 * @export
 */
const ClipsApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.ClipsApiFp)(configuration);
    return {
        /**
         *
         * @summary Create a clip
         * @param {CreateClipRequest} createClipRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createClip(createClipRequest, options) {
            return localVarFp.createClip(createClipRequest, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Delete a specific clip
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteClip(id, options) {
            return localVarFp.deleteClip(id, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get driver list of an actor
         * @param {string} id
         * @param {number} [limit] pagination - the amount of actors to return
         * @param {string} [token] pagination - the pagination token from the previous response, default &#x3D; 100
         * @param {*} [options] Override http request option.
         * @deprecated
         * @throws {RequiredError}
         */
        getActorDrivers(id, limit, token, options) {
            return localVarFp.getActorDrivers(id, limit, token, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get actors list
         * @param {number} [limit] pagination - the amount of actors to return
         * @param {string} [token] pagination - the pagination token from the previous response, default &#x3D; 100
         * @param {*} [options] Override http request option.
         * @deprecated
         * @throws {RequiredError}
         */
        getActors(limit, token, options) {
            return localVarFp.getActors(limit, token, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get a specific clip
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getClip(id, options) {
            return localVarFp.getClip(id, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get clips
         * @param {number} [limit] pagination - the amount of clips to return
         * @param {string} [token] pagination - the pagination token from the previous response, default &#x3D; 100
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getClips(limit, token, options) {
            return localVarFp.getClips(limit, token, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get presenter by id
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getPresenterById(id, options) {
            return localVarFp.getPresenterById(id, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get presenter by id
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getPresenterById_1(id, options) {
            return localVarFp.getPresenterById_1(id, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get presenter drivers by id
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getPresenterDriversById(id, options) {
            return localVarFp.getPresenterDriversById(id, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get presenter drivers by id
         * @param {string} id
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getPresenterDriversById_2(id, options) {
            return localVarFp.getPresenterDriversById_2(id, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get a list of the presenters
         * @param {number} [limit] pagination - the amount of presenters to return
         * @param {string} [token] pagination - the pagination token from the previous response, default &#x3D; 100
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getPresenters(limit, token, options) {
            return localVarFp.getPresenters(limit, token, options).then((request) => request(axios, basePath));
        },
        /**
         *
         * @summary Get a list of the presenters
         * @param {number} [limit] pagination - the amount of presenters to return
         * @param {string} [token] pagination - the pagination token from the previous response, default &#x3D; 100
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getPresenters_3(limit, token, options) {
            return localVarFp.getPresenters_3(limit, token, options).then((request) => request(axios, basePath));
        },
    };
};
exports.ClipsApiFactory = ClipsApiFactory;
/**
 * ClipsApi - object-oriented interface
 * @export
 * @class ClipsApi
 * @extends {BaseAPI}
 */
class ClipsApi extends base_1.BaseAPI {
    /**
     *
     * @summary Create a clip
     * @param {CreateClipRequest} createClipRequest
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ClipsApi
     */
    createClip(createClipRequest, options) {
        return (0, exports.ClipsApiFp)(this.configuration).createClip(createClipRequest, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Delete a specific clip
     * @param {string} id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ClipsApi
     */
    deleteClip(id, options) {
        return (0, exports.ClipsApiFp)(this.configuration).deleteClip(id, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get driver list of an actor
     * @param {string} id
     * @param {number} [limit] pagination - the amount of actors to return
     * @param {string} [token] pagination - the pagination token from the previous response, default &#x3D; 100
     * @param {*} [options] Override http request option.
     * @deprecated
     * @throws {RequiredError}
     * @memberof ClipsApi
     */
    getActorDrivers(id, limit, token, options) {
        return (0, exports.ClipsApiFp)(this.configuration).getActorDrivers(id, limit, token, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get actors list
     * @param {number} [limit] pagination - the amount of actors to return
     * @param {string} [token] pagination - the pagination token from the previous response, default &#x3D; 100
     * @param {*} [options] Override http request option.
     * @deprecated
     * @throws {RequiredError}
     * @memberof ClipsApi
     */
    getActors(limit, token, options) {
        return (0, exports.ClipsApiFp)(this.configuration).getActors(limit, token, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get a specific clip
     * @param {string} id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ClipsApi
     */
    getClip(id, options) {
        return (0, exports.ClipsApiFp)(this.configuration).getClip(id, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get clips
     * @param {number} [limit] pagination - the amount of clips to return
     * @param {string} [token] pagination - the pagination token from the previous response, default &#x3D; 100
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ClipsApi
     */
    getClips(limit, token, options) {
        return (0, exports.ClipsApiFp)(this.configuration).getClips(limit, token, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get presenter by id
     * @param {string} id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ClipsApi
     */
    getPresenterById(id, options) {
        return (0, exports.ClipsApiFp)(this.configuration).getPresenterById(id, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get presenter by id
     * @param {string} id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ClipsApi
     */
    getPresenterById_1(id, options) {
        return (0, exports.ClipsApiFp)(this.configuration).getPresenterById_1(id, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get presenter drivers by id
     * @param {string} id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ClipsApi
     */
    getPresenterDriversById(id, options) {
        return (0, exports.ClipsApiFp)(this.configuration).getPresenterDriversById(id, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get presenter drivers by id
     * @param {string} id
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ClipsApi
     */
    getPresenterDriversById_2(id, options) {
        return (0, exports.ClipsApiFp)(this.configuration).getPresenterDriversById_2(id, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get a list of the presenters
     * @param {number} [limit] pagination - the amount of presenters to return
     * @param {string} [token] pagination - the pagination token from the previous response, default &#x3D; 100
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ClipsApi
     */
    getPresenters(limit, token, options) {
        return (0, exports.ClipsApiFp)(this.configuration).getPresenters(limit, token, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     *
     * @summary Get a list of the presenters
     * @param {number} [limit] pagination - the amount of presenters to return
     * @param {string} [token] pagination - the pagination token from the previous response, default &#x3D; 100
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof ClipsApi
     */
    getPresenters_3(limit, token, options) {
        return (0, exports.ClipsApiFp)(this.configuration).getPresenters_3(limit, token, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.ClipsApi = ClipsApi;
